Python/Product/Debugger/Debugger/DebugConnectionListener.cs

```diff
@@ -95,39 +98,64 @@ static class DebugConnectionListener {
             var stream = new NetworkStream(socket, ownsSocket: true);
             try {
                 socket.Blocking = true;
-                string debugId = stream.ReadString();
-                var result = (ConnErrorMessages)stream.ReadInt32();
 
-                lock (_targets) {
-                    Guid debugGuid;
-                    WeakReference weakProcess;
-                    PythonProcess targetProcess;
-
-                    if (Guid.TryParse(debugId, out debugGuid) &&
-                        _targets.TryGetValue(debugGuid, out weakProcess) &&
-                        (targetProcess = weakProcess.Target as PythonProcess) != null) {
-
-                        if (result == ConnErrorMessages.None) {
-                            targetProcess.Connect(stream);
-                            stream = null;
-                            socket = null;
+                var debugConn = new DebugConnection(stream);
+                try {
+                    string debugId = string.Empty;
+                    var result = ConnErrorMessages.None;
+                    using (var connectedEvent = new AutoResetEvent(false)) {
+                        EventHandler<LDP.LocalConnectedEvent> eventReceived = (object sender, LDP.LocalConnectedEvent ea) => {
+                            result = (ConnErrorMessages)ea.result;
+                            debugId = ea.processGuid;
+                            try {
+                                connectedEvent.Set();
+                            } catch (ObjectDisposedException) {
+                            }
+                            debugConn.Authenticated();
+                        };
```

Comment:
Should this also unsubscribe itself? Even if it doesn't get invoked twice, it's a closure that'll have a bunch of references...